src/staticanalysis/bot/tests/test_clang.py

```diff
@@ -148,9 +148,11 @@ def test_clang_tidy(mock_repository, mock_config, mock_clang, mock_stats, mock_r
     assert len(issues) == 2
     assert isinstance(issues[0], ClangTidyIssue)
     assert issues[0].check == 'modernize-use-nullptr'
+    assert issues[0].reason == 'Modernize our code base to C++11'
     assert issues[0].line == 3
     assert isinstance(issues[1], ClangTidyIssue)
     assert issues[1].check == 'modernize-use-nullptr'
+    assert issues[1].reason == 'Modernize our code base to C++11'
```

Comment:
Bonus point if you add another case in the unit test to test a check *without* a reason (and check that `reason is None`)

Reply:
Hey @La0 with the test. How to trigger a clang tidy error in using BAD_SRC_CPP. Can you take a look at irc Please.